src/csharp/Grpc.Tools.CSharpGenerator/Program.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+
+namespace Grpc.Tools.CSharpGenerator
+{
+    class Program
+    {
+        //From Managed.Windows.Forms/XplatUI
+        [DllImport("libc")]
+        static extern int uname(IntPtr buf);
+        static bool IsRunningOnMac()
+        {
+            var buf = IntPtr.Zero;
+            try
+            {
+                buf = Marshal.AllocHGlobal(8192);
+                // This is a hacktastic way of getting sysname from uname ()
+                if (uname(buf) == 0)
+                {
+                    var os = Marshal.PtrToStringAnsi(buf);
+                    if (os == "Darwin")
+                        return true;
+                }
+            }
+            catch
+            {
+            }
+            finally
+            {
+                if (buf != IntPtr.Zero)
+                    Marshal.FreeHGlobal(buf);
+            }
+            return false;
+        }        
+        static void Main(string[] args)
+        {
+            try
+            {
+                var projectDir = args[0];
```

Comment:
I think it might be ok to have this depend on Grpc.Core - because this is a build time only dependency. Unfortunately this class is internal so I can't use it as-is. Would you feel better if I used some other nuget package that included this check? Again, note that this a build-time dependency only.